### ISSUE TITLE
generic_operation test fails randomly when dtype is specified

### ISSUE THREAD

**Issue by scheherzade** [STEllAR-GROUP/phylanx#771]
The following PhySL code returns one of these two outputs randomly each time you run it:
```
cout(floor__int(10.2))
```
either returns ``` 10 ```, or
```
floor__int:: unknown function requested: floor__int: HPX(bad_parameter)
```
Same thing happens for other functions in generic_operation when dtype is specified. 
```
cout(absolute__float(-10.2))
```
either returns ``` 10.2 ```, or
```
absolute__float:: unknown function requested: absolute__float: HPX(bad_parameter)
```

**Comment by khuck** [comment]
This is related to issue #767

**Comment by hkaiser** [comment]
this was fixed by merging #774
